### PR TITLE
Add ip-ranges to find()

### PR DESCRIPTION
**Pull request by Miosame**
I'm not sure if the way the servers array is filled aligns with the general idea and what should happen if a wrong range is passed, so that's up to discussion / review process.

Currently it:
- checks inside `findLocalDevices()` if it is a range and assigns / fills the servers array based on that.
- returns the error, which should probably be made into a throw instead (or an empty `return` to return undefined, similar to how it is done throughout other parts of the code?) together with notes inside the readme if it _will_ throw instead of returning undefined.

Again: Edits are checked / allowed and welcome.

### CODE REVIEW

File: __tests__/index.js [hunk 1]
```diff
@@ -31,14 +31,39 @@ describe('local-devices', () => {
 
       it('returns the result of all IPs', async () => {
         const result = await find()
+        console.log(platform, result)
```
Comment (natterstefan) [inline __tests__/index.js]:
```suggestion
```

Please, remove the console log here.

File: src/index.js [hunk 2]
```diff
@@ -22,12 +23,34 @@ const options = {
...
+    }
+  } else {
+    servers = getServers()
+  }
+
+  if (!lock[key]) {
+    if (address) {
+      if (new RegExp('/|-').test(key)) {
+        // range of IPs detected
+        lock[key] = arpAll().then(unlock(key))
```
Comment (natterstefan) [inline src/index.js]:
```suggestion
        lock[key] = pingServers().then(arpAll).then(unlock(key))
```

Don't we need to invoke `pingServers` (Sends a ping to all servers to update the arp table.) first, as well here?

Then you could simplify this if-block, as the `// no address given` does the same thing.

File: src/index.js [hunk 3]
```diff
@@ -22,12 +23,34 @@ const options = {
...
 module.exports = function findLocalDevices (address) {
   var key = String(address)
-  lock[key] =
-    lock[key] ||
-    (address
-      ? pingServer(address).then(arpOne)
-      : pingServers().then(arpAll)
-    ).then(unlock(key))
+
+  if (address && new RegExp('/|-').test(address)) {
```
Comment (natterstefan) [inline src/index.js]:
```js
new RegExp('/|-').test(address)
```

It is used twice. You could create a helper fn for that.

**Comment by Miosame** [comment]
Thanks! will take a look at the rest once I come home later tonight.

File: src/index.js (hunk 2 above)
Comment (Miosame) [reply src/index.js]: Indeed: https://github.com/DylanPiercey/local-devices/blob/a27a69d1779ea7d2d77cddabeaeb722e70ccd8fc/src/index.js#L29

**Comment by Miosame** [comment]
Done, implemented the suggestions and cleaned up a little.

**Comment by DylanPiercey** [comment]
It would be nice if when a range is used that it only ping'd servers in that range. Currently the most costly part of this module is pinging all of the servers and ranges could help with that.

**Comment by Miosame** [comment]
@DylanPiercey that already should be taken care of by:

https://github.com/Miosame/local-devices/blob/1d642f9a140bdff79e831f97bd1fedc83390074e/src/index.js#L27-L37

and

https://github.com/Miosame/local-devices/blob/1d642f9a140bdff79e831f97bd1fedc83390074e/src/index.js#L77-L82

I believe?

**Comment by DylanPiercey** [comment]
@Miosame ah yes, sorry I missed that.

**Comment by Miosame** [comment]
@DylanPiercey no worries, I had to check too first 👍

**Comment by Miosame** [comment]
@DylanPiercey I'm currently also under heavy load, but just remembered about this - is there anything that worries you before the merge we could look at?

**Comment by Miosame** [comment]
@DylanPiercey bump? :tada:

**Comment by DylanPiercey** [comment]
@Miosame could you add an extra example to the docs demonstrating the range functionality? I think it looks good to merge otherwise.

**Comment by natterstefan** [comment]
Good point @DylanPiercey. Can you take care of this @Miosame, please?

**Comment by Miosame** [comment]
@natterstefan @DylanPiercey sure will do this later today or tomorrow :+1:

**Comment by Miosame** [comment]
@natterstefan @DylanPiercey done :tada:

File: README.md [hunk 4]
```diff
@@ -52,6 +52,30 @@ find('192.168.0.10').then(device => {
...
+find('192.168.0.0/24').then(devices => {
+    devices /*
+    [
+        { name: '?', ip: '192.168.0.10', mac: '...' },
+        { name: '...', ip: '192.168.0.50', mac: '...' },
+        { name: '...', ip: '192.168.0.155', mac: '...' },
+        { name: '...', ip: '192.168.0.211', mac: '...' }
+    ]
+    */
+})
```
Comment (natterstefan) [inline README.md:78]: Please adjust the indentation, so it aligns with the other examples of the README.

```suggestion
// Find all devices within 192.168.0.1 to 192.168.0.25 range
find('192.168.0.1-192.168.0.25').then(devices => {
    devices /*
    [
      { name: '?', ip: '192.168.0.10', mac: '...' },
      { name: '...', ip: '192.168.0.17', mac: '...' },
      { name: '...', ip: '192.168.0.21', mac: '...' },
      { name: '...', ip: '192.168.0.22', mac: '...' }
    ]
    */
})

// Find all devices within /24 subnet range of 192.168.0.x
find('192.168.0.0/24').then(devices => {
    devices /*
    [
      { name: '?', ip: '192.168.0.10', mac: '...' },
      { name: '...', ip: '192.168.0.50', mac: '...' },
      { name: '...', ip: '192.168.0.155', mac: '...' },
      { name: '...', ip: '192.168.0.211', mac: '...' }
    ]
    */
})
```

**Review by natterstefan — APPROVED** [review]
One more thing, then it's okay for me. Thanks a lot!

**Comment by Miosame** [comment]
@natterstefan didn't notice that locally, thanks!

**Pull request merged**